### PR TITLE
Fixes #15257 - let print adapters decide whether to paginate

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

**Comment by tstrachota** [comment]
Small update: I removed the default pagination setting from the config file template.

**Comment by akofink** [comment]
Looks great; I like the design. The tests pass locally and seem to cover the functionality implemented.

ACK from me, though I'm not in the Foreman org

File: lib/hammer_cli/output/adapter/abstract.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@ def initialize(context={}, formatters={})
       @formatters = HammerCLI::Output::Formatters::FormatterLibrary.new(filter_formatters(formatters))
     end
 
+    def paginate_by_default?
+      true
+    end
+
```
Comment (mbacovsky) [inline lib/hammer_cli/output/adapter/abstract.rb]: Nit: I'd prefer to refactor to
```ruby 
def paginate_by_default?
    !!@paginate_by_default
end
```
and set the attribute in initializer, but it is minor thing

**Comment by tstrachota** [comment]
@mbacovsky ok, updated

**Comment by mbacovsky** [comment]
:+1: Thanks @tstrachota! :poodle:

**Pull request merged**